docs/lib/default.nix

```diff
@@ -0,0 +1,103 @@
+# Generates the documentation for library functions using nixdoc.
+# See https://github.com/nix-community/nixdoc
+{
+  lib,
+  runCommand,
+  writers,
+  nixdoc,
+  nixvim,
+  pageSpecs ? (
+    lib.pipe ../../lib/docs.toml [
+      lib.importTOML
+      (toml: toml.page or [ ])
+      (builtins.map (page: page // { file = ../../lib/${page.file}; }))
+    ]
+  ),
+}:
+
+let
+  locToFile = loc: "lib/${lib.strings.concatStringsSep "/" loc}.md";
+  locToName = lib.attrsets.showAttrPath;
+in
+
+runCommand "nixvim-lib-docs"
+  {
+    nativeBuildInputs = [
+      nixdoc
+    ];
+
+    locations = writers.writeJSON "locations.json" (
+      import ./function-locations.nix {
+        inherit lib;
+        rootPath = nixvim;
+        functionSet = lib.extend nixvim.lib.overlay;
+        pathsToScan = builtins.catAttrs "loc" pageSpecs;
+        revision = nixvim.rev or "main";
+      }
+    );
+
+    passthru.menu = ''
+      # Library functions
+
+      - [`lib`](${locToFile [ "index" ]})
```

Comment:
Originally, I was allowing for having `lib.nixvim.*` and also a `lib.types` page, so I made the root node `lib` instead of `lib.nixvim`.

![image](https://github.com/user-attachments/assets/7fa7d4cf-8436-4f4c-b3f8-aa72b1b148ab)

In hindsight, maybe that's confusing? Especially as the "index" page is _really_ about using `lib.nixvim` not `lib`...